laddertools/test_glicko.py

```diff
@@ -0,0 +1,32 @@
+import pytest
+from .ranking import RankingGlicko, _RatingGlicko
+from .ladder import _Player
+
+import numpy
+
+
+def test_glicko():
+    # Based on example from https://www.glicko.net/glicko/glicko2.pdf
+    me = _RatingGlicko(1500, 200)
+    opponents = [
+        _RatingGlicko(1400, 30),
+        _RatingGlicko(1550, 100),
+        _RatingGlicko(1700, 300),
+    ]
+    outcomes = [1, 0, 0]  # I won, then lost two
+
+    new_rating = RankingGlicko.compute_new_rating(
+        me, opponents, outcomes, tau=0.5, eps=1e-6
+    )
+
+    numpy.testing.assert_almost_equal(1464.06, new_rating.r, decimal=2)
+    numpy.testing.assert_almost_equal(151.52, new_rating.RD, decimal=2)
+    numpy.testing.assert_almost_equal(0.05999, new_rating.std, decimal=5)
+
+
+def test_glicko_empty_period_will_increase_RD():
+    me = _RatingGlicko(1500, 345, std=1.0)
+    opponents = outcomes = []
+    new_rating = RankingGlicko.compute_new_rating(me, opponents, outcomes)
+    assert new_rating.RD == 350
+    assert new_rating.r == me.r  # should remain unchanged... Only increase RD
```

Comment:
```
============================================= test session starts =============================================
platform linux -- Python 3.9.1, pytest-6.2.1, py-1.10.0, pluggy-0.13.1
rootdir: /home/ux/src/oraladder
collected 2 items

laddertools/test_glicko.py FF                                                                           [100%]

================================================== FAILURES ===================================================
_________________________________________________ test_glicko _________________________________________________

    def test_glicko():
        # Based on example from https://www.glicko.net/glicko/glicko2.pdf
>       me = _RatingGlicko(1500, 200)
E       TypeError: __init__() missing 1 required positional argument: 'std'

laddertools/test_glicko.py:10: TypeError
__________________________________ test_glicko_empty_period_will_increase_RD __________________________________

    def test_glicko_empty_period_will_increase_RD():
        me = _RatingGlicko(1500, 345, std=1.0)
        opponents = outcomes = []
        new_rating = RankingGlicko.compute_new_rating(me, opponents, outcomes)
>       assert new_rating.RD == 350
E       AttributeError: 'tuple' object has no attribute 'RD'

laddertools/test_glicko.py:31: AttributeError
=========================================== short test summary info ===========================================
FAILED laddertools/test_glicko.py::test_glicko - TypeError: __init__() missing 1 required positional argumen...
FAILED laddertools/test_glicko.py::test_glicko_empty_period_will_increase_RD - AttributeError: 'tuple' objec...
============================================== 2 failed in 0.91s ==============================================
```

:(

BTW, I think this may be useful:

```diff
diff --git a/Makefile b/Makefile
index 5341f3d..248a9ca 100644
--- a/Makefile
+++ b/Makefile
@@ -37,6 +37,9 @@ $(LADDER_MAP_PACK): $(VENV)
 $(RAGL_MAP_PACK): $(VENV)
        (. $(ACTIVATE) && ora-mapstool misc/map-pools/ragl-s10.maps --pack $(RAGL_MAP_PACK))

+test: $(VENV)
+       (. $(ACTIVATE) && pytest -v)
+
 clean:
        $(RM) -r build
        $(RM) -r dist
@@ -48,4 +51,4 @@ $(VENV):
        $(PYTHON) -m venv $@
        ( . $(ACTIVATE) && pip install -e .)

-.PHONY: ladderdev initladderdev wheel clean mappacks ragldev initragldev
+.PHONY: ladderdev initladderdev wheel clean mappacks ragldev initragldev test
```